chain/network/src/peer_manager/connection/mod.rs

```diff
@@ -109,6 +110,46 @@ impl Connection {
         self.addr.do_send(SendMessage { message: msg }.with_span_context());
     }
 
+    pub fn send_routing_table_update(
+        self: &Arc<Self>,
+        rtu: Arc<RoutingTableUpdate>,
+    ) -> impl Future<Output = ()> {
+        let this = self.clone();
+        async move {
+            let res = this
+                .send_routing_table_update_demux
+                .call(rtu, {
+                    let this = this.clone();
+                    |rtus: Vec<Arc<RoutingTableUpdate>>| async move {
+                        this.send_message(Arc::new(PeerMessage::SyncRoutingTable(
```

Comment:
let's mayeb extract a function here to avoid train-wreck code layout?

Reply:
done